vnext/Microsoft.ReactNative/Views/FlyoutViewManager.cpp

```diff
@@ -384,7 +384,16 @@ void FlyoutShadowNode::SetTargetFrameworkElement() {
       }
     }
   } else {
-    m_targetElement = xaml::Window::Current().Content().as<xaml::FrameworkElement>();
+    if (IsXamlIsland()) {
+      // // XamlRoot added in 19H1
+      if (Is19H1OrHigher()) {
+        if (auto xamlRoot = React::XamlUIService::GetXamlRoot(GetViewManager()->GetReactContext().Properties())) {
+          m_targetElement = xamlRoot.Content().try_as<xaml::FrameworkElement>();
```

Comment:
I think we closed this PR too eagerly. I solved a slightly different issue. We still need to set the target element from the XamlRoot for XAML Islands. @lyahdav - could you re-open this one?